data-prepper-logstash-configuration/src/main/java/org/opensearch/dataprepper/logstash/mapping/CsvLogstashPluginAttributesMapper.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.logstash.mapping;
+
+import org.opensearch.dataprepper.logstash.model.LogstashAttribute;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * LogstashPluginAttributesMapper that maps Logstash's csv filter to Data Prepper's csv processor.
+ * In addition to the settings in csv.mapping.yaml, this mapper supports Logstash's autogenerate_column_names setting.
+ *
+ */
+public class CsvLogstashPluginAttributesMapper extends AbstractLogstashPluginAttributesMapper{
+    protected static final String LOGSTASH_AUTOGENERATE_COLUMN_NAMES_ATTRIBUTE_NAME = "autogenerate_column_names";
+    protected static final String LOGSTASH_COLUMNS_ATTRIBUTE_NAME = "columns";
+    protected static final String DATA_PREPPER_COLUMN_NAMES = "column_names";
+
+    @Override
+    protected void mapCustomAttributes(final List<LogstashAttribute> logstashAttributes,
+                                       final LogstashAttributesMappings logstashAttributesMappings,
+                                       final Map<String, Object> pluginSettings) {
+
+
+        final Optional<LogstashAttribute> autogenerateColumnNamesAttribute = findLogstashAttribute(logstashAttributes,
+                LOGSTASH_AUTOGENERATE_COLUMN_NAMES_ATTRIBUTE_NAME);
+        Object autogenerateColumnNamesValue = autogenerateColumnNamesAttribute.map(attribute -> attribute.getAttributeValue().getValue())
+                .orElse(false);
+        boolean isAutogenerateColumnNames = autogenerateColumnNamesValue.equals(true);
+
+        final Optional<LogstashAttribute> columnsAttribute = findLogstashAttribute(logstashAttributes,
+                LOGSTASH_COLUMNS_ATTRIBUTE_NAME);
+        Object columnsValue = columnsAttribute.map(attribute -> attribute.getAttributeValue().getValue()).orElse(false);
+        boolean columnsValueIsEmptyOrDoesNotExist = columnsValue.equals(false) || ((List<String>) columnsValue).isEmpty();
```

Comment:
I think this can be simplified to something like
```
final List<String> columnsValue = columnsAttribute.map(attribute -> attribute.getAttributeValue().getValue()).orElse(Collections.emptyList());
final boolean columnsValueIsEmptyOrDoesNotExist = columnValues.isEmpty();
```
Using `Object` to get around being false or a list is kind of hacky

Reply:
Thanks for calling this out!